lib/project_types/script/layers/domain/metadata.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Domain
+      class Metadata
+        attr_reader :schema_major_version, :schema_minor_version
+
+        def initialize(schema_major_version, schema_minor_version)
+          @schema_major_version = schema_major_version
+          @schema_minor_version = schema_minor_version
+        end
+
+        class << self
+          def create_from_json(metadata_json)
+            metadata_hash = JSON.parse(metadata_json)
+            schema_versions = metadata_hash["schemaVersions"]
+            if schema_versions.nil?
+              err_msg = "Metadata is missing schemaVersions"
```

Comment:
Generally we catch errors and specify error output in https://github.com/Shopify/shopify-app-cli/blob/c619af5936d3baddd07143342a4b9cb00d8533d7/lib/project_types/script/ui/error_handler.rb#L163 which include a help suggestion sometimes. It makes it easier for folks like Kate to update error messages that are potentially user-facing and we don't show a stacktrace to end-users.

Also we should reconsider the actual error message content, I just put these together quickly so I could assert things were being sent correctly and get something I could understand if they weren't being picked up.

Reply:
Localization error handling updated in latest commit; will consult with @kwringe on exact text